src/Storage/Adapters/WindowsAzure.Storage.9/AzureContextAdapterExtensions.cs

```diff
@@ -0,0 +1,135 @@
+
+namespace Microsoft.WindowsAzure.Commands.Storage.Adapters
+{
+    public static class AzureContextAdapterExtensions
+    {
+        /// <summary>
+        /// Get the current storage account.
+        /// </summary>
+        /// <param name="context">The current Azure context.</param>
+        /// <returns>The current storage account, or null, if no current storage account is set.</returns>
+        public static CloudStorageAccount GetCurrentStorageAccount(this IAzureContext context, IStorageServiceProvider provider)
+        {
+            if (context != null)
+            {
+                var storageConnectionString = context.GetCurrentStorageAccountConnectionString();
+                try
+                {
+                    return
+                        CloudStorageAccount.Parse(storageConnectionString);
+                }
+                catch
+                {
+                    var service = provider.GetStorageService(storageConnectionString, null);
+                    return service.GetCloudStorageAccount();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Get a CloudStorageAccount client for the storage account represented by this context
+        /// </summary>
+        /// <param name="context">The storage context for the storage account</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static CloudStorageAccount GetCloudStorageAccount(this IStorageContext context)
+        {
+            CloudStorageAccount result = null;
+            CloudStorageAccount.TryParse(context.ConnectionString, out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Get a CloudStorageAccount client for the storage account represented by this service
+        /// </summary>
+        /// <param name="service">The storage service</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static CloudStorageAccount GetCloudStorageAccount( this IStorageService service)
+        {
+            return new CloudStorageAccount(new StorageCredentials(service.Name, service.AuthenticationKeys.First()),
+                new StorageUri(service.BlobEndpoint), new StorageUri(service.QueueEndpoint), 
+                new StorageUri(service.TableEndpoint), new StorageUri(service.FileEndpoint));
+        }
+
+        /// <summary>
+        /// Get a storage context client for the storage account represented by this service
+        /// </summary>
+        /// <param name="service">The storage service</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static IStorageContext GetStorageContext(this IStorageService service)
+        {
+            return new AzureStorageContext(new CloudStorageAccount(new StorageCredentials(service.Name, service.AuthenticationKeys.First()),
```

Comment:
We need to make sure this fills out the connection string and all endpoints (even if these are filled with dummy values) included in IStorageContext